### PR TITLE
Recipient Filter Update

### PR DESCRIPTION
**Pull request by alburde1**
- Updating recipient filter to use the exact text sent instead of a list of `legal_entity` IDs

### CODE REVIEW

File: usaspending_api/awards/v2/filters/transaction.py [hunk 1]
```diff
@@ -171,6 +171,7 @@ def transaction_filter(filters):
                     awarding_agency__subtier_agency__name__in=awarding_subtier
                 )
 
+        # TODO update this to recipient_search_text if we ever switch back from the matview
```
Comment (tony-sappe) [inline usaspending_api/awards/v2/filters/transaction.py]: AFAIK this is still used by bulk downloader, it isn't dead code yet

File: usaspending_api/awards/v2/filters/award.py [hunk 2]
```diff
@@ -171,6 +171,7 @@ def award_filter(filters):
             if len(awarding_subtier) != 0:
                 queryset &= Award.objects.filter(awarding_agency__subtier_agency__name__in=awarding_subtier)
 
+        # TODO update this to recipient_search_text if we ever switch back from the matview
```
Comment (nmonga91) [inline usaspending_api/awards/v2/filters/award.py]: We should update this now

File: usaspending_api/awards/v2/filters/transaction.py (hunk 1 above)
Comment (nmonga91) [reply usaspending_api/awards/v2/filters/transaction.py]: We should update this here now as well

File: usaspending_api/awards/v2/filters/matview_transaction.py [hunk 3]
```diff
@@ -115,7 +115,7 @@ def transaction_filter(filters):
                     'time_period',
                     'award_type_codes',
                     'agencies',
-                    'legal_entities',
```
Comment (tony-sappe) [inline usaspending_api/awards/v2/filters/matview_transaction.py]: In the interest of not breaking anything, could we keep the existing filter inplace and add the new search_text field? Are we 100% sure nothing will use `legal_entities` after the website PR?

Comment (nmonga91) [reply usaspending_api/awards/v2/filters/matview_transaction.py]: I would think we shouldn't allow it going forward. @tony-sappe if we're not 100% sure, then I agree, we should keep it.

Comment (willkjackson) [reply usaspending_api/awards/v2/filters/matview_transaction.py]: Keep the filter.

**Pull request merged**